src/app/modules/test-case-service/default.test.case.service.ts

```diff
@@ -0,0 +1,81 @@
+import { TestCaseServiceConfig } from './test.case.service.config';
+import { Injectable, Injector } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { MessagingService } from '@testeditor/messaging-service';
+
+// code duplication with persistence service and test-editor-web, removal planned with next refactoring
+const HTTP_CLIENT_NEEDED = 'httpClient.needed';
+const HTTP_CLIENT_SUPPLIED = 'httpClient.supplied';
+
+// API provided by service backend
+export class CallTreeNode {
+  displayName: string;
+  children: CallTreeNode[];
+}
+
+export abstract class TestCaseService {
+  abstract getCallTree(path: string, onResponse?: (some: any) => void, onError?: (error: any) => void): void;
+}
+
+
+@Injectable()
+export class DefaultTestCaseService extends TestCaseService {
+
+  private static readonly callTreeURLPath = '/call-tree';
+  private serviceUrl: string;
+
+  private cachedHttpClient: HttpClient;
+
+  constructor(config: TestCaseServiceConfig, private messagingService: MessagingService) {
+    super();
+    this.serviceUrl = config.testCaseServiceUrl;
+  }
+
+  getCallTree(path: string,
+              onResponse?: (status: CallTreeNode) => void,
+              onError?: (error: any) => void): void {
+    console.log('DefaultTestCaseService.getCallTree');
+    this.httpClientExecute(httpClient => {
+      console.log('got http client');
+      const requestUrl =  `${this.serviceUrl}${DefaultTestCaseService.callTreeURLPath}?resource=${encodeURIComponent(path)}`;
+      console.log('request: ' + requestUrl);
+      return httpClient.get<CallTreeNode>(requestUrl)
+        .toPromise();
```

Comment:
didn't we want to transition away from promises in favor of observables…?

Reply:
We should think about that again. Since I do not return any promise out of this class, I would not gain anything using observables here...